### ISSUE TITLE
Filler stock price data being fetched for the "end" parameter 

### ISSUE THREAD

**Issue by rodrigobercini** [dpguthrie/yahooquery#18]
It's me again, and again, thank you so much for developing this amazing API.

**Describe the bug**
When requesting intraday stock prices, the rows up to the "end" date parameter are getting filled with duplicated data.

**How To Reproduce**
1) Fetch ticker data.
```
from yahooquery import Ticker
aapl = Ticker('AAPL')
```

2) Check for intraday stock prices.

`aapl.history(interval='15m', start='2020-05-12', end='2020-05-14')`

3) Notice the issue.
![image](https://user-images.githubusercontent.com/18664907/83907146-4887f480-a73b-11ea-82cb-4e551569f170.png)

**Expected behavior**
1. Query stock price starting 2020-05-13 to 2020-05-14
2. Get only the data from 2020-05-13 to 2020-05-14 without filler data

**Desktop:**
- Google Colab Notebooks

**Additional context**
I think this has to do with the fact that every intraday stock price query returns the current stock price, which must mess with how the dataframe is returned.

**Comment by dpguthrie** [comment]
Thanks for this issue too.  I think you're right.  For whatever reason, the API has to return the current stock price and will return a bunch of nulls in between the end date and current.  The other issue you raised will play a part in this as well.  I think a solution could be to just return a subset of the dataframe based on the data returned from the request:

```
ls = data[symbol]['indicators']['quote'][0]['close']
idx = next((i for i, v in enumerate(ls) if v is None), None)
if idx:
    return df[:idx]
return df
```

Let me know if you have any other suggestions.

**Comment by rodrigobercini** [comment]
Your solution makes a lot of sense, I don't see why someone querying a specific timeframe would need the most up-to-date price. And even if they do need it, they can fetch it via another query.

An alternative solution would be slicing the dataframe using the parameters passed in the query:

`aapl.history(interval='15m', start=start, end=end)`

`    return df[start:end]
`
Does that make any sense?
Cheers!

**Comment by dpguthrie** [comment]
That makes sense.  Much cleaner solution than the one I proposed.  I might also just try `df.dropna(inplace=True)`.  But, it seems like this behaviour only occurs when the market is open as I'm unable to replicate right now.

**Comment by aironsholt** [comment]
First, thanks for a great module.

I've seen the same/similar problem with the ticker GLIF which has only been traded a few times a seen here (data from the yfinance module):
                            Open   High    Low  Close  Volume  Dividends  Stock Splits
Datetime
2020-07-06 13:17:00-04:00  23.81  23.82  23.81  23.82     320          0             0
2020-07-07 10:51:00-04:00  23.70  23.70  23.49  23.49       0          0             0
2020-07-09 11:26:00-04:00  23.26  23.26  23.26  23.26       0          0             0
2020-07-09 15:24:00-04:00  23.26  23.26  23.25  23.26     399          0             0


But yahooquery fills in a lot of non-existing data. I tracked down the issue and it seems to be this piece of the code that fills in the gaps (line 1025ff in ticker.py):
```
            try:
                df[columns] = df.groupby(['symbol'])[columns].ffill()
            except (KeyError, ValueError):
                df.fillna(method='ffill', inplace=True)
```
Also notice, that using ffill makes the summation of the volumes wrong (read: so it doesn't match the daily volume)

Instead I'll suggest, that  you simply remove the 'nan' data points, as there are not data :-)

**Comment by dpguthrie** [comment]
This was addressed in 2.2.5.  Let me know if you see any issues with the implementation.  Thank you both for your help.

**Comment by rodrigobercini** [comment]
This issue is also solved.

Thank you!
